<h1>Class ClassNotFoundException</h1>   <ul class="inheritance"> <li><a href="object">java.lang.Object</a></li> <li> <ul class="inheritance"> <li><a href="throwable">java.lang.Throwable</a></li> <li> <ul class="inheritance"> <li><a href="exception">java.lang.Exception</a></li> <li> <ul class="inheritance"> <li><a href="reflectiveoperationexception">java.lang.ReflectiveOperationException</a></li> <li> <ul class="inheritance"> <li>java.lang.ClassNotFoundException</li> </ul> </li> </ul> </li> </ul> </li> </ul> </li> </ul>   <dl> <dt>All Implemented Interfaces:</dt> <dd><a href="../io/serializable">Serializable</a></dd> </dl>   <pre data-language="java">public class ClassNotFoundException
extends ReflectiveOperationException</pre> <div>
<p>Thrown when an application tries to load in a class through its string name using: <ul> <li>The <code>forName</code> method in class <code>Class</code>. </li>
<li>The <code>findSystemClass</code> method in class <code>ClassLoader</code> . </li>
<li>The <code>loadClass</code> method in class <code>ClassLoader</code>. </li>
</ul> </p>
<p> but no definition for the class with the specified name could be found. </p>
<p>As of release 1.4, this exception has been retrofitted to conform to the general purpose exception-chaining mechanism. The "optional exception that was raised while loading the class" that may be provided at construction time and accessed via the <a href="classnotfoundexception#getException--"><code>getException()</code></a> method is now known as the <i>cause</i>, and may be accessed via the <a href="throwable#getCause--"><code>Throwable.getCause()</code></a> method, as well as the aforementioned "legacy method."</p>
</div> <dl> <dt>Since:</dt> <dd>JDK1.0</dd> <dt>See Also:</dt> <dd>
<a href="class#forName-java.lang.String-"><code>Class.forName(java.lang.String)</code></a>, <a href="classloader#findSystemClass-java.lang.String-"><code>ClassLoader.findSystemClass(java.lang.String)</code></a>, <a href="classloader#loadClass-java.lang.String-boolean-"><code>ClassLoader.loadClass(java.lang.String, boolean)</code></a>, Serialized Form</dd> </dl>       <h2 id="constructor.detail">Constructors</h2>   <h3 id="ClassNotFoundException--">ClassNotFoundException</h3> <pre data-language="java">public ClassNotFoundException()</pre> <p>Constructs a <code>ClassNotFoundException</code> with no detail message.</p>    <h3 id="ClassNotFoundException-java.lang.String-">ClassNotFoundException</h3> <pre data-language="java">public ClassNotFoundException(String s)</pre> <p>Constructs a <code>ClassNotFoundException</code> with the specified detail message.</p> <dl> <dt>Parameters:</dt> <dd>
<code>s</code> - the detail message.</dd> </dl>    <h3 id="ClassNotFoundException-java.lang.String-java.lang.Throwable-">ClassNotFoundException</h3> <pre data-language="java">public ClassNotFoundException(String s,
                              Throwable ex)</pre> <p>Constructs a <code>ClassNotFoundException</code> with the specified detail message and optional exception that was raised while loading the class.</p> <dl> <dt>Parameters:</dt> <dd>
<code>s</code> - the detail message</dd> <dd>
<code>ex</code> - the exception that was raised while loading the class</dd> <dt>Since:</dt> <dd>1.2</dd> </dl>     <h2 id="method.detail">Methods</h2>   <h3 id="getException--">getException</h3> <pre data-language="java">public Throwable getException()</pre> <div>
<p>Returns the exception that was raised if an error occurred while attempting to load the class. Otherwise, returns <code>null</code>. </p>
<p>This method predates the general-purpose exception chaining facility. The <a href="throwable#getCause--"><code>Throwable.getCause()</code></a> method is now the preferred means of obtaining this information.</p>
</div> <dl> <dt>Returns:</dt> <dd>the <code>Exception</code> that was raised while loading a class</dd> <dt>Since:</dt> <dd>1.2</dd> </dl>    <h3 id="getCause--">getCause</h3> <pre data-language="java">public Throwable getCause()</pre> <p>Returns the cause of this exception (the exception that was raised if an error occurred while attempting to load the class; otherwise <code>null</code>).</p> <dl> <dt>Overrides:</dt> <dd>
<code><a href="throwable#getCause--">getCause</a></code> in class <code><a href="throwable">Throwable</a></code>
</dd> <dt>Returns:</dt> <dd>the cause of this exception.</dd> <dt>Since:</dt> <dd>1.4</dd> </dl>     <div class="_attribution">
  <p class="_attribution-p">
    &copy; 1993&ndash;2017, Oracle and/or its affiliates. All rights reserved.<br>Documentation extracted from Debian's OpenJDK Development Kit package.<br>Licensed under the GNU General Public License, version 2, with the Classpath Exception.<br>Various third party code in OpenJDK is licensed under different licenses (see Debian package).<br>Java and OpenJDK are trademarks or registered trademarks of Oracle and/or its affiliates.<br>
    
  </p>
</div>
